icu4c/source/common/ucurr.cpp

```diff
@@ -805,21 +796,19 @@ ucurr_getPluralName(const char16_t* currency,
     rb = ures_getByKey(rb, CURRENCYPLURALS, rb, &ec2);
 
     // Fetch resource with multi-level resource inheritance fallback
-    rb = ures_getByKeyWithFallback(rb, buf, rb, &ec2);
+    LocalUResourceBundlePointer curr(ures_getByKeyWithFallback(rb, buf, rb, &ec2));
 
-    s = ures_getStringByKeyWithFallback(rb, pluralCount, len, &ec2);
+    s = ures_getStringByKeyWithFallback(curr.getAlias(), pluralCount, len, &ec2);
     if (U_FAILURE(ec2)) {
         //  fall back to "other"
         ec2 = U_ZERO_ERROR;
-        s = ures_getStringByKeyWithFallback(rb, "other", len, &ec2);     
+        s = ures_getStringByKeyWithFallback(curr.getAlias(), "other", len, &ec2);     
         if (U_FAILURE(ec2)) {
-            ures_close(rb);
             // fall back to long name in Currencies
             return ucurr_getName(currency, locale, UCURR_LONG_NAME, 
                                  isChoiceFormat, len, ec);
         }
     }
-    ures_close(rb);
```

Comment:
Is it really correct to now never close `rb`?

Reply:
destrutor of curr will close it